File: leetcode3026/solveLeetcode3026.cpp
// class Solution {
// public:
//     long long maximumSubarraySum(vector<int>& nums, int k) 
//     {
        
//         unordered_map<int,long long>minprefixSum;

//         long long prefix=0,ans=LLONG_MIN;
        
//         for(const auto&x:nums)
//         {


//             if(minprefixSum.count(x))
//                 minprefixSum[x]=min(minprefixSum[x],prefix);
//             else minprefixSum[x]=prefix;

//             prefix+=x;

//             if(minprefixSum.count(x-k))
//                 ans=max(ans,prefix-minprefixSum[x-k]);
//             if(minprefixSum.count(x+k))
//                 ans=max(ans,prefix-minprefixSum[x+k]);

//         }

//         return ans>LLONG_MIN?ans:0;
//     }
// };

File: leetcode3026/solveLeetcode3026.go
// func maximumSubarraySum(nums []int, k int) int64 {
// 	minPrefixSum := make(map[int]int64)
// 	prefix := int64(0)
// 	var ans int64= math.MinInt64

// 	for _, x := range nums {
// 		if val, ok := minPrefixSum[x]; ok {
// 			minPrefixSum[x] = min(val, prefix)
// 		} else {
// 			minPrefixSum[x] = prefix
// 		}

// 		prefix += int64(x)

// 		if val, ok := minPrefixSum[x-k]; ok {
// 			ans = max(ans, prefix-val)
// 		}
// 		if val, ok := minPrefixSum[x+k]; ok {
// 			ans = max(ans, prefix-val)
// 		}
// 	}

// 	if ans == math.MinInt64 {
// 		return 0
// 	}
// 	return ans
// }